File: golang/Hash/groupAnagrams/groupAnagrams.go
package groupanagrams

import (
	"slices"
)

func groupAnagrams(strs []string) [][]string {

	anagramsMap := make(map[string][]string)
	for _, str := range strs {
		b := []byte(str)
		// sort.Slice(b, func(i, j int) bool { return b[i] < b[j] })
		slices.Sort(b)
		word := string(b)
		anagramsMap[word] = append(anagramsMap[word], str)
	}
	res := [][]string{}
	for _, anagram := range anagramsMap {
		res = append(res, anagram)
	}
	return res
}

File: java/Hash/groupAnagrams/groupAnagrams.java
import java.util.*;

class groupAnagrams {

    public List<List<String>> groupAnagram(String[] strs) {
        Map<String, List<String>> anagramMap = new HashMap<>();

        for (int i = 0; i < strs.length; i++) {
            char[] word = strs[i].toCharArray();
            Arrays.sort(word);
            String stringWord = new String(word);
            List<String> list = anagramMap.get(stringWord);
            if (list == null) {
                list = new ArrayList<>();
                anagramMap.put(stringWord, list);
            }
            list.add(strs[i]);
        }
        return new ArrayList<>(anagramMap.values());
    }
}
